flashinfer/cute_dsl/blockwise_gemm.py

```diff
@@ -0,0 +1,2955 @@
+                a_data_ptr,
+                b_data_ptr,
+                sfa_data_ptr,
+                sfb_data_ptr,
+                c_data_ptr,
+            ) = (
+                a_tensor_gpu.data_ptr(),
+                b_tensor_gpu.data_ptr(),
+                sfa_tensor_gpu.data_ptr(),
+                sfb_tensor_gpu.data_ptr(),
+                c_tensor_gpu.data_ptr(),
+            )
+
+        a_ptr = make_ptr(
+            ab_dtype,
+            a_data_ptr,
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        b_ptr = make_ptr(
+            ab_dtype,
+            b_data_ptr,
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        sfa_ptr = make_ptr(
+            sf_dtype,
+            sfa_data_ptr,
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        sfb_ptr = make_ptr(
+            sf_dtype,
+            sfb_data_ptr,
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        c_ptr = make_ptr(
+            c_dtype,
+            c_data_ptr,
+            cute.AddressSpace.gmem,
+            assumed_align=16,
+        )
+        return [a_ptr, b_ptr, sfa_ptr, sfb_ptr, c_ptr]
+
+    kernel = cute.compile(
+        BlockwiseGemmCuteDSL(
+            m=m,
+            n=n,
+            k=k,
+            l=l,
+            a_major=a_major,
+            b_major=b_major,
+            c_major=c_major,
+            ab_dtype=ab_dtype,
+            sf_dtype=sf_dtype,
+            c_dtype=c_dtype,
+            acc_dtype=acc_dtype,
+            use_2cta_instrs=use_2cta_instrs,
+            mma_tiler_mn=mma_tiler_mn,
+            cluster_shape_mn=cluster_shape_mn,
+        ),
+        *get_cute_pointers(None),
+        cutlass_torch.current_stream(),
+    )
+
+    def tensor_api(
+        a_tensor_gpu: torch.Tensor,
+        b_tensor_gpu: torch.Tensor,
+        sfa_tensor_gpu: torch.Tensor,
+        sfb_tensor_gpu: torch.Tensor,
+        c_tensor_gpu: Optional[torch.Tensor] = None,
+    ):
+        if c_tensor_gpu is None:
+            c_tensor_gpu = torch.empty(
+                (l, m, n),
+                dtype=cutlass_to_torch_dtype(c_dtype),
+                device="cuda",
+            )
+
+        current_stream = cutlass_torch.current_stream()
+
+        nonlocal kernel
+        kernel(
+            *get_cute_pointers(
+                [
+                    a_tensor_gpu,
+                    b_tensor_gpu,
+                    sfa_tensor_gpu,
+                    sfb_tensor_gpu,
+                    c_tensor_gpu,
+                ]
+            ),
+            current_stream,
+        )
+        return c_tensor_gpu
+
+    return tensor_api
+
+
+def blockwise_gemm(
+    a_torch: torch.Tensor,
+    sfa_torch: torch.Tensor,
+    b_torch: torch.Tensor,
+    sfb_torch: torch.Tensor,
+    c_torch: torch.Tensor,
+    *,
+    ab_dtype: str,
+    sf_dtype: str,
+    c_dtype: str,
+    acc_dtype: str,
+    **kwargs,
```

Comment:
Is this intended to be a user-facing interface, or will it eventually be called internally by something like `fp8_mm`?
I’m asking because we may want to follow a consistent naming convention for the API and its parameters. (@yzh119 do we have one?)
I understand that the backend will eventually add more features such as communication, but it would still be good to have a well-defined interface at this stage.
Also, adding a docstring to clarify the behavior of this function would be helpful.